cloudevents/extensions/tracing.md

```diff
@@ -0,0 +1,208 @@
+# Event Tracing
+
+This extension defines attributes for tracking event relationships and causality in distributed systems, enabling comprehensive traceability through correlation and causation identifiers.
+
+## Notational Conventions
+
+As with the main [CloudEvents specification](../spec.md), the key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+## Attributes
+
+### correlationid
+
+- Type: `String`
+- Description: An identifier that groups related events within the same logical flow or business transaction. All events sharing the same correlation ID are part of the same workflow.
+- Constraints
+  - OPTIONAL
+  - If present, MUST be a non-empty string
+
+### causationid
+
+- Type: `String`
+- Description: The unique identifier of the event that directly caused this event to be generated. This SHOULD be the `id` value of the causing event.
+- Constraints
+  - OPTIONAL
+  - If present, MUST be a non-empty string
+
+## Usage
+
+The Event Tracing extension provides two complementary mechanisms for tracking event relationships:
+
+1. **Correlation ID**: Groups all events that are part of the same logical flow, regardless of their causal relationships
+2. **Causation ID**: Tracks the direct parent-child relationships between events in a causal chain
+
+These attributes can be used independently or together, depending on the tracing requirements of your system.
+
+### Correlation vs Causation
+
+Understanding the distinction between these two concepts is crucial:
+
+- **Correlation ID** answers: "Which events are part of the same business transaction?"
+- **Causation ID** answers: "Which specific event directly triggered this event?"
```

Comment:
maybe change the first "event" to "occurrence" to avoid it sounding a bit recursive. Plus we talk about events by saying:
```
An "event" is a data record expressing an occurrence and its context.
```